### PR TITLE
[master < T945] Add periodic iterate docs

### PR DESCRIPTION
**Pull request by Josipmrden**
### Description

This is a new query module which helps users executed difficult and complex queries by batching results of one query into another.

### Pull request type

Please delete options that are not relevant and check the ones that are.

- [X] New documentation page 

### Checklist:

- [X] I checked all content with Grammarly
- [X] I performed a self-review of my code
- [X] I made corresponding changes to the rest of the documentation
- [X] The build passes locally
- [X] My changes generate no new warnings or errors

### CODE REVIEW

File: mage/query-modules/cpp/periodic.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  padding: '0.2rem',
+}}>
+{children}
+</span>
+);
+
+**Periodic module** enables users to execute a query periodically in batches. Periodic in this case
+does not mean that the query is executed after a specific timeout. It actually signifies that the results
+of some input source is being batched in order to speed up the execution time, due to the complexity of the
+query.
```
Comment (vpavicic) [inline mage/query-modules/cpp/periodic.md:26]:
```suggestion
The **periodic module** enables users to execute a query periodically in batches. In this case, the name periodic doesn't indicate that the query is executed after a time interval, but rather that, due to the complexity of the query, the results of some input source are batched to speed up execution.
```

File: mage/query-modules/cpp/periodic.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+:::
+
+
+
+#### Input:
+
+- `input_query: string` ➡ the input query which will yield the results that need to be batched
+- `running_query: string` ➡ query which will be executed on the batched results
+- `params: Map[string, string]` ➡ parameters for the procedure
+  - `batch_size: Integer` ➡ key specifying for many results should be kept in a batch
```
Comment (vpavicic) [inline mage/query-modules/cpp/periodic.md:53]:
```suggestion
  - `batch_size: Integer` ➡ key specifying how many results should be kept in a batch 
```

File: mage/query-modules/cpp/periodic.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+is committed by itself. An issue while running this procedure will not result in rollbacking batches of committed results.
+:::
+
+
+
+#### Input:
+
+- `input_query: string` ➡ the input query which will yield the results that need to be batched
+- `running_query: string` ➡ query which will be executed on the batched results
+- `params: Map[string, string]` ➡ parameters for the procedure
```
Comment (vpavicic) [inline mage/query-modules/cpp/periodic.md:52]: there will be more of them in the future?

**Review by vpavicic — CHANGES_REQUESTED** [review]
some lang changes + a Q about the warning

File: mage/query-modules/cpp/periodic.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+### Procedures
+
+### `iterate(input_query, running_query, params)`
+
+:::caution
+This module breaks the isolation levels in the database, as the results are batched and executed in different transactions. Do
+not consider the query which executes this procedure to be aligned with set isolation level of the database, as every executed batch
+is committed by itself. An issue while running this procedure will not result in rollbacking batches of committed results.
+:::
```
[the remark this answers is not in the document]
Comment (Josipmrden) [reply mage/query-modules/cpp/periodic.md:44]: There is only one procedure here. So I'd say both hold true

File: mage/query-modules/cpp/periodic.md (hunk 3 above)
Comment (Josipmrden) [reply mage/query-modules/cpp/periodic.md:52]: not sure

Comment (Josipmrden) [reply mage/query-modules/cpp/periodic.md:52]: possibly, but currently no need

File: mage/query-modules/cpp/periodic.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vpavicic) [reply mage/query-modules/cpp/periodic.md:46]: If there are currently no plans to expand this module with more procedures, i would just write the whole warning in the intro if that's ok with you :) (i can do it, just give me the thumbs up)

**Review by vpavicic — APPROVED** [review]

**Pull request merged**